clippy_lints/src/methods/mod.rs

```diff
@@ -107,6 +108,29 @@ declare_clippy_lint! {
     "used `cloned` where `copied` could be used instead"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `_.cloned().last()`.
+    ///
+    /// ### Why is this bad?
+    /// Performance, clone only need to be executed once.
+    ///
+    /// ### Example
+    /// ```rust
+    /// # let vec = vec!["string".to_string()];
+    /// vec.iter().cloned().last();
+    /// ```
+    /// Could be written as
+    /// ```rust
+    /// # let vec = vec!["string".to_string()];
+    /// vec.iter().last().cloned();
+    /// ```
+    #[clippy::version = "1.59.0"]
+    pub CLONED_LAST,
+    complexity,
```

Comment:
I think this should be `perf` or `nursery`, depending on the number of additional warnings in lintcheck vs. master. If it's too noisy, I'd prefer the latter, otherwise I'd look into the added warnings and see if the benefit outweighs the added effort to fix the lint.

Reply:
@llogiq It looks to me that maybe the lint should be split. Sometimes it may not make a sense to do it, if it's just `.cloned().next()`, though it depend on whats after.

Other times it may be worth it, because you may avoid cloning the whole collection.

Reply:
I was planning to write a more complicated check, which can cover more cases, but I think this PR is a good start. There are a few things that I don't know how to use yet.

Reply:
I don't think that would warrant a split: Just because the code has `.next()` now doesn't mean that it won't get changed to `.nth(_)` later. Also even when only changing `.cloned().next()`, to `.next().cloned()`, this will reduce compile time if ever so slightly, so that's a perf benefit anyway.